podpac/core/node.py

```diff
@@ -974,7 +979,14 @@ def node_eval(fn):
     cache_key = "output"
 
     @functools.wraps(fn)
-    def wrapper(self, coordinates, output=None):
+    def wrapper(self, coordinates, output=None, selector=None):
+        # check crs compatibility
+        if (output is not None) and ("crs" in output.attrs) and (output.attrs["crs"] != coordinates.crs):
+            raise ValueError(
+                "Output coordinate reference system ({}) does not match".format(output.crs)
+                + "request Coordinates coordinate reference system ({})".format(coordinates.crs)
+            )
+
```

Comment:
Ah, this is a great place for this check.

Reply:
I was pretty pleased when I discovered this one. :)